### PR TITLE
feat(union-type): adds support for OneOf and AnyOf

### LINKED ISSUE #9: Add support for Union Types
This task is related to adding support for union types (i.e. `OneOf` and `AnyOf` cases) that are supported in OpenApi specifications. More detail can be found [here](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/). The coverage should be complete for the following cases
 - Collection (Array, Map, Array Of Map & Map of Array)
 - Non-Collection (Custom types w/ discriminators, Native types, Enums, Date & DateTime, etc)

### PR DESCRIPTION
**Pull request by hamzashoukat94**
This PR supports union types that are OneOf and AnyOf types by providing the base type by the name of the UnionType which is an abstract class. The concrete implementation of this class is OneOf, AnyOf, and LeafType classes which are part of the core library.

closes #9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,12 +24,13 @@ gem 'apimatic_core_interfaces'
...
+| Name                                                                                 | Description                                                                               |
+|--------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------|
+| [`HttpClient`](lib/apimatic-core-interfaces/client/http_client.rb)                   | To save both Request and Response after the completion of response                        |
+| [`ResponseFactory`](lib/apimatic-core-interfaces/factories/response_factory.rb)      | To convert the client-adapter response into a custom HTTP response                        |
+| [`Authentication`](lib/apimatic-core-interfaces/types/authentication.rb)             | To setup methods for the validation and application of the required authentication scheme |
+| [`UnionType`](lib/apimatic-core-interfaces/types/union_type.rb)                      | To setup methods for the validation and deserialization of OneOf/AnyOf union types   |
```
Comment (sufyankhanrao) [inline README.md:32]: Blocker: Serialization section needs to be included in the description since we are now handling the serialization as well.

**Review by sufyankhanrao — CHANGES_REQUESTED** [review]
Above mentioned change needs your attention.

File: README.md (hunk 1 above)
Comment (hamzashoukat94) [reply README.md]: Done

**Review by sufyankhanrao — DISMISSED** [review]
LGTM

**Review by sufyankhanrao — APPROVED** [review]

**Pull request merged**